apps/zotonic_launcher/src/zotonic.erl

```diff
@@ -112,9 +112,25 @@ stop() ->
         end,
         ok,
         Sites),
+    
     % Wait a bit till all sites are stopped (max 5 secs)
     await_sites_stopping(50),
-    init:stop(0).
+
+    % Stop all other running applications. Note: on OTP 25 we can simply use init:stop(0).
+    % On earlier OTP versions this is problematic because all modules are unloaded before
+    % the node is stopped. This can take a long time.
+    application:stop(zotonic_launcher),
+    application:stop(exometer),
+    application:stop(jobs),
+    application:stop(mnesia),
+    application:stop(epgsql),
+
+    [ application:stop(A) || {A, _, _} <- application:which_applications() ],
```

Comment:
Should we first stop the other applications and then the base apps like exometer, jobs and mnesia?

It might be that the other apps want to save something in the databases.

Reply:
Ah, yes you are right. I noticed the order is important. When it is wrong it can lead to unexpected hangs. Maybe we should just use `init:stop/0`. OTP 25 is just around the corner for anybody who complains about slowness.

Reply:
It is _really_ slow.

We might also use `init:stop/0` if the OTP version is >= 25.

And the other shutdown on older versions.